File: language-support/js/daml-ledger/tests/arbitrary/completion_stream_request.ts
// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

import * as jsc from 'jsverify';
import * as ledger from '../../src';
import { LedgerOffset } from './ledger_offset';

export const CompletionStreamRequest: jsc.Arbitrary<ledger.CompletionStreamRequest> =
    jsc.tuple([jsc.string, LedgerOffset, jsc.array(jsc.string)]).smap<ledger.CompletionStreamRequest>(
        ([applicationId, offset, parties]) => ({
            applicationId: applicationId,
            offset: offset,
            parties: parties
        }),
        (request) =>
            [request.applicationId, request.offset, request.parties]
    );

File: sdk/canton/community/ledger/ledger-api-core/src/main/scala/com/digitalasset/canton/ledger/api/messages/command/completion/CompletionStreamRequest.scala
// Copyright (c) 2025 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

package com.digitalasset.canton.ledger.api.messages.command.completion

import com.digitalasset.canton.data.Offset
import com.digitalasset.daml.lf.data.Ref

final case class CompletionStreamRequest(
    userId: Ref.UserId,
    parties: Set[Ref.Party],
    offset: Option[Offset],
)
